.github/workflows/build_and_test.yml

```diff
@@ -2,6 +2,13 @@ name: Build and test
 on: [push, pull_request]
 
 jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+    - uses: pre-commit/action@v2.0.3
```

Comment:
Let's not include pre-commit checks in the CI yet, otherwise it will keep failing until we resolve all issues. Let me open PRs to fix all issues, and then we can include this in the CI

Reply:
Since we are planning to add I only comment out these lines, so we can add easily later